### PR TITLE
Branch validation

### PR DESCRIPTION
**Pull request by stoivo**

### CODE REVIEW

File: core/commands/checkout.py [hunk 1]
```diff
@@ -53,10 +54,17 @@ class GsCheckoutNewBranchCommand(WindowCommand, GitCommand):
...
-    def run_async(self):
-        self.window.show_input_panel(NEW_BRANCH_PROMPT, "", self.on_done, None, None)
+    def run_async(self, name=""):
+        self.window.show_input_panel(NEW_BRANCH_PROMPT, name, self.on_done, None, None)
 
     def on_done(self, branch_name):
+        pattern = r"^(?!\.|.*\.\..*|.*@.*|\/)[a-zA-Z0-9\-\_\/\.]+(?<!\.lock)(?<!\/)(?<!\.)\b$"
+        match = re.match(pattern, branch_name)
+        if not match:
+            sublime.status_message("`{}` is a invalid branch name.".format(branch_name))
```
Comment (divmain) [inline core/commands/checkout.py]: Please change this to an error modal message - the status message bar can be easy to miss.

**Comment by divmain** [comment]
Yikes, that regex is getting gnarly.  I think this is fine for now, though.  Can you rebase on top of current master, as well as the minor change?  As soon as I get the green button, I'll merge this in.  Thanks!

**Comment by stoivo** [comment]
I am sorry, for so many commits. I tried squash them.

**Comment by divmain** [comment]
No worries, this works.  Thanks again!

**Comment by stoivo** [comment]
Hi, if i have some more thing i would like to add to GitSavvy. What is the prosess for that? Should i just create a new branch on my fork and make an other PR. Do I need to wright up what I want to work on somewhere?

**Comment by divmain** [comment]
If you're thinking of making significant changes, you can open an issue and discuss ahead of time.  You can also create a new branch (make sure you're branching off of `divmain/master`) and submit a PR.  There's no guarantee I'll accept it, but sometimes it is easier to demonstrate what you're thinking with code.

**Pull request merged**